[CI] EnrichRestIT test {p0=enrich/30_tsdb_index/enrich documents over _bulk} failing

### CI Link

https://gradle-enterprise.elastic.co/s/q2dskwszr5pbc/tests/:x-pack:plugin:enrich:qa:rest:yamlRestTest/org.elasticsearch.xpack.enrich.EnrichRestIT/test%20%7Bp0=enrich%2F30_tsdb_index%2Fenrich%20documents%20over%20_bulk%7D?top-execution=1

### Repro line

`./gradlew ':x-pack:plugin:enrich:qa:rest:yamlRestTest' --tests "org.elasticsearch.xpack.enrich.EnrichRestIT.test {p0=enrich/30_tsdb_index/enrich documents over _bulk}" -Dtests.seed=D33E918FE8B20189 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=de-LU -Dtests.timezone=America/Inuvik -Druntime.java=17`	

### Does it reproduce?

Didn't try

### Applicable branches

main

### Failure history

https://gradle-enterprise.elastic.co/scans/tests?search.timeZoneId=Europe/Zurich&tests.container=org.elasticsearch.xpack.enrich.EnrichRestIT&tests.test=test%20%7Bp0%3Denrich/30_tsdb_index/enrich%20documents%20over%20_bulk%7D

### Failure excerpt

```
[2022-08-26T01:29:24,712][INFO ][o.e.x.e.EnrichRestIT     ] [test] Stash dump on test failure [{
  "stash" : {
    "body" : {
      "error" : {
        "root_cause" : [
          {
            "type" : "illegal_argument_exception",
            "reason" : "unknown setting [index.mode] did you mean [index.codec]?",
            "stack_trace" : "org.elasticsearch.ElasticsearchException$1: unknown setting [index.mode] did you mean [index.codec]?
	at org.elasticsearch.server@8.5.0/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:640)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:568)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.RestResponse.build(RestResponse.java:143)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:102)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:81)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:55)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.client.internal.node.NodeClient$SafelyWrappedActionListener.onFailure(NodeClient.java:170)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:216)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:275)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.ActionListener$DelegatingActionListener.onFailure(ActionListener.java:218)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.ActionRunnable.onFailure(ActionRunnable.java:77)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:28)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:241)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.doStart(TransportMasterNodeAction.java:279)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:216)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:59)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:86)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.ActionFilter$Simple.apply(ActionFilter.java:53)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:84)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:61)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.tasks.TaskManager.registerAndExecute(TaskManager.java:201)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.client.internal.node.NodeClient.executeLocally(NodeClient.java:112)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.client.internal.node.NodeClient.doExecute(NodeClient.java:90)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:380)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.client.internal.support.AbstractClient$IndicesAdmin.execute(AbstractClient.java:1271)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.client.internal.support.AbstractClient$IndicesAdmin.create(AbstractClient.java:1346)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.action.admin.indices.RestCreateIndexAction.lambda$prepareRequest$0(RestCreateIndexAction.java:57)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.BaseRestHandler.handleRequest(BaseRestHandler.java:103)
	at org.elasticsearch.security@8.5.0/org.elasticsearch.xpack.security.rest.SecurityRestFilter.handleRequest(SecurityRestFilter.java:119)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:395)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.RestController.tryAllHandlers(RestController.java:529)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:310)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.http.AbstractHttpServerTransport.dispatchRequest(AbstractHttpServerTransport.java:375)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.http.AbstractHttpServerTransport.handleIncomingRequest(AbstractHttpServerTransport.java:456)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.http.AbstractHttpServerTransport.incomingRequest(AbstractHttpServerTransport.java:349)
	at org.elasticsearch.transport.netty4@8.5.0/org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.handlePipelinedRequest(Netty4HttpPipeliningHandler.java:112)
	at org.elasticsearch.transport.netty4@8.5.0/org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(Netty4HttpPipeliningHandler.java:102)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.codec@4.1.77.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.codec@4.1.77.Final/io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.codec@4.1.77.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.codec@4.1.77.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.codec@4.1.77.Final/io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:327)
	at io.netty.codec@4.1.77.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:299)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler@4.1.77.Final/io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.codec@4.1.77.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.transport@4.1.77.Final/io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.transport@4.1.77.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.transport@4.1.77.Final/io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.transport@4.1.77.Final/io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.transport@4.1.77.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
	at io.netty.transport@4.1.77.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:623)
	at io.netty.transport@4.1.77.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:586)
	at io.netty.transport@4.1.77.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
	at io.netty.common@4.1.77.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995)
	at io.netty.common@4.1.77.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: unknown setting [index.mode] did you mean [index.codec]?
	at org.elasticsearch.server@8.5.0/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:559)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:505)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:475)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:445)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.normalizeRequestSetting(MetadataCreateIndexService.java:320)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.onlyCreateIndex(MetadataCreateIndexService.java:287)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.createIndex(MetadataCreateIndexService.java:255)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.admin.indices.create.TransportCreateIndexAction.masterOperation(TransportCreateIndexAction.java:148)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.admin.indices.create.TransportCreateIndexAction.masterOperation(TransportCreateIndexAction.java:44)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.master.TransportMasterNodeAction.executeMasterOperation(TransportMasterNodeAction.java:128)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$3(TransportMasterNodeAction.java:279)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62)
	at org.elasticsearch.server@8.5.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	... 66 more
	Suppressed: java.lang.IllegalArgumentException: unknown setting [index.routing_path] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
		... 79 more
"
          }
        ],
```

Another failure [here](https://gradle-enterprise.elastic.co/s/5qtfmzy4gqfec) with a similar (but not the same) error message
```
java.lang.AssertionError: Failure at [enrich/30_tsdb_index:3]: expected [2xx] status code but api [indices.create] returned [400 Bad Request] [--- |  
-- | --
  | error: |  
  | root_cause: |  
  | - type: "illegal_argument_exception" |  
  | reason: "unknown setting [index.routing_path] please check that any required plugins\ |  
  | \ are installed, or check the breaking changes documentation for removed settings" |  
  | stack_trace: "org.elasticsearch.ElasticsearchException$1: unknown setting [index.routing_path]\ |  
  | \ please check that any required plugins are installed, or check the breaking\ |  
  | \ changes documentation for removed settings\n\tat org.elasticsearch.server@8.5.0/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:640)\n\ |  
  | \tat org.elasticsearch.server@8.5.0/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:568)\n\
```

The test failed 10 times in the past 7 days.

--- a/x-pack/plugin/enrich/qa/rest/build.gradle
+++ b/x-pack/plugin/enrich/qa/rest/build.gradle
@@ -2,6 +2,8 @@ apply plugin: 'elasticsearch.internal-java-rest-test'
 apply plugin: 'elasticsearch.internal-yaml-rest-test'
 apply plugin: 'elasticsearch.yaml-rest-compat-test'
 
+
+import org.elasticsearch.gradle.Version
 import org.elasticsearch.gradle.internal.info.BuildParams
 
 restResources {
@@ -28,4 +30,5 @@ testClusters.configureEach {
   setting 'xpack.license.self_generated.type', 'basic'
   setting 'xpack.monitoring.collection.enabled', 'true'
   setting 'xpack.security.enabled', 'false'
+  requiresFeature 'es.index_mode_feature_flag_registered', Version.fromString("8.4.0")
 }

